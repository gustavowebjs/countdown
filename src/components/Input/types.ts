export type InputProps = {
  label?: string;
  name?: string;
  type?: string;
  placeholder?: string;
  autoComplete?: string;
  value?: string;
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
  length?: number;
};
