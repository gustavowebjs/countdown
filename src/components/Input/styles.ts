import styled from 'styled-components';

export const Container = styled.div`
  width: 100%;
`;

export const Input = styled.input`
  background: #ffffff;
  border: 2px solid #cccccc;
  border-radius: 6px;
  height: 42px;

  font-size: 14px;
  font-weight: 400;
  line-height: 28px;
  color: ${({ theme }) => theme.colors.darker};
  padding: 16px 10px;
  width: 100%;

  :focus {
    border: 2px solid ${({ theme }) => theme.colors.primary};
  }
`;
