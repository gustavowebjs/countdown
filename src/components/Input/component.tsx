import React from 'react';
import * as T from './types';
import * as S from './styles';
import Label from '../Label';

const Input: React.ForwardRefRenderFunction<HTMLInputElement, T.InputProps> = (
  props,
  ref
) => {
  const {
    name,
    value,
    placeholder,
    label,
    type,
    autoComplete,
    length,
    onChange,
  } = props;

  return (
    <S.Container>
      {label && <Label text={label} />}

      <S.Input
        autoComplete={autoComplete}
        type={type}
        name={name}
        maxLength={length}
        value={value}
        onChange={onChange}
        ref={ref}
        placeholder={placeholder}
      />
    </S.Container>
  );
};

export default React.forwardRef<HTMLInputElement, T.InputProps>(Input);
