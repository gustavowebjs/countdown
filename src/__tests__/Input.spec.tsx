import { shallow } from 'enzyme';
import Input from '../components/Input';

describe('Input Component', () => {
  it('Should input component render correctly', () => {
    const wrapper = shallow(<Input className="input" placeholder="Email" />);

    expect(wrapper.find('.input').length).toEqual(1);
    expect(wrapper.find('.input').prop('placeholder')).toEqual('Email');
    expect(wrapper).toMatchSnapshot();
  });
});
