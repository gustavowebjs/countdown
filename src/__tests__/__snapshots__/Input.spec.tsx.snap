// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component Should input component render correctly 1`] = `
<styled.div>
  <styled.input
    className="input"
    placeholder="Email"
  />
</styled.div>
`;
